src/systems/filecoin_markets/storage_market/storage_client.id

```diff
@@ -1,18 +1,62 @@
 import ipld "github.com/filecoin-project/specs/libraries/ipld"
-import deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
-import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+import deal "github.com/filecoin-project/specs/systems/filecoin_markets/storage_market/storage_deal"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+import libp2p "github.com/filecoin-project/specs/libraries/libp2p"
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+
+type ClientDeal struct {
+    ProposalCid     ipld.CID
+    Proposal        deal.StorageDealProposal
+    State           StorageDealStatus
+    Miner           libp2p.PeerID
+    MinerWorker     addr.Address
+    DealID          deal.DealID
+    PublishMessage  ipld.CID
+}
+
+type ProposeStorageDealResult struct {
+    ProposalCid ipld.CID
+}
+
+type StorageProviderInfo struct {
+    Address     addr.Address  // actor address
+    Owner       addr.Address
+    Worker      addr.Address  // signs messages
```

Comment:
likewise worker @anorth ? can't it be derived from the actor addr, assuming we mean the storage market actor?